File: 412. Fizz Buzz/412. Fizz Buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    var i = 1, fizz = 1, buzz = 1, res = [];
    
    while(i <= n) {
        if(fizz === 3 && buzz === 5) {
            res.push("FizzBuzz");
            fizz = 1, buzz = 1;
        } else if (fizz === 3) {
            res.push("Fizz");
            fizz = 1;
            buzz++;
        } else if (buzz === 5) {
            res.push("Buzz");
            buzz = 1;
            fizz++;
        } else {
            res.push(i.toString());
            buzz++;
            fizz++
        }
        i++;
    }
    
    return res;
};

File: 412. Fizz Buzz/412. Fizz Buzz.rb
# @param {Integer} n
# @return {String[]}
def fizz_buzz(n)
    i = 1
    res = []
    while i <= n
        if (i % 3 == 0) && (i % 5 == 0)
            res << "FizzBuzz"
        elsif i % 3 == 0
            res << "Fizz"
        elsif i % 5 == 0
            res << "Buzz"
        else
            res << i.to_s
        end
        i += 1
    end
    res
end
